### PR TITLE
RAC-9: add a simple empty tab with an association picker

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 1]
```diff
@@ -212,16 +219,87 @@ define([
...
-          if (associationTypes.length) {
+          if (0 !== associationTypes.length) {
             const currentGrid = this.datagrids[this.getCurrentAssociationTarget()];
             this.renderGrid(currentGrid.name, currentGrid.getInitialParams(this.getCurrentAssociationType()));
             this.setListenerSelectors();
           }
 
+          const quantifiedAssociationTypes = associationTypes.filter(({code}) => 'PACK' === code)
+          const associations = {
+            PACK: {
```
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:235]: this is temporary to make it work. Will be removed in the next PRs

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 2]
```diff
@@ -260,7 +343,7 @@ define([
      * Refresh the associations panel after model change
      */
     postUpdate: function() {
-      if (this.isVisible()) {
+      if (this.isVisible() && !this.$el.is(':empty')) {
```
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:346]: avoid double rendering at product loading

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 3]
```diff
@@ -344,6 +427,15 @@ define([
       const associationType = event.currentTarget.dataset.associationType;
       this.setCurrentAssociationType(associationType);
 
+      const isQuantifiedAssociation = this.isQuantifiedAssociation(state.associationTypes, associationType);
+      if (isQuantifiedAssociation) {
+        this.$('.association-grid-container').addClass('hide');
```
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:432]: yes, this is crapy, but we are trying to add a flower on top of a pile of poop so it's the best we can do ^^

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations/quantified-associations.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+type AssociationType = {
+  code: string;
+};
+
+type QuantifiedLink = {
```
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations/quantified-associations.tsx:7]: will be moved in next PR to a proper model file

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js (hunk 1 above)
Comment (SamirBoulil) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:235]: (you can add a TODO: TO remove if you want)

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/templates/product/tab/association-panes.html [hunk 5]
```diff
@@ -1,49 +1,59 @@
 <% _.each(associationTypes, function (associationType) { %>
-    <div class="AknTabHeader association-type<% if (associationType.code !== currentAssociationType) { %> AknTitleContainer--hidden<% } %>" data-association-type="<%- associationType.code %>">
```
Comment (SamirBoulil) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/templates/product/tab/association-panes.html:2]: this is just reindent right ?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 6]
```diff
@@ -212,16 +219,48 @@ define([
...
+          if (0 !== associationTypes.length) {
             const currentGrid = this.datagrids[this.getCurrentAssociationTarget()];
             this.renderGrid(currentGrid.name, currentGrid.getInitialParams(this.getCurrentAssociationType()));
             this.setListenerSelectors();
           }
 
+          const quantifiedAssociationTypes = associationTypes.filter(({code}) => 'PACK' === code)
+          const associations = {
+            PACK: {
+              products: [{identifier: 'bag', quantifty: '25'}],
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:236]: quantifty ? :smile:

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 7]
```diff
@@ -396,6 +451,18 @@ define([
...
+
+      isQuantifiedAssociation ?
+        this.$('#product-quantified-associations').removeClass('hide') :
+        this.$('#product-quantified-associations').addClass('hide')
+    },
+
+    isQuantifiedAssociation: function (associationTypes, associationTypeCode) {
+      const associationType =  associationTypes.find(associationType => associationType.code === associationTypeCode);
+
+      if (undefined === associationType) throw new Error(`Cannot find assocation type ${associationTypeCode}`);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:463]:
```suggestion
      if (undefined === associationType) throw new Error(`Cannot find association type ${associationTypeCode}`);
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 8]
```diff
@@ -742,6 +809,47 @@ define([
...
+      }, previousQuantifiedLinkedProducts);
+
+      const linkedProductModels = productAndProductModelIdentifiers.reduce((linkedProductModels, identifier) => {
+        const matchProductModel = identifier.match(/^product_model_(.*)$/);
+        if (!matchProductModel) {
+          return linkedProductModels;
+        }
+
+        const productModelIdentifierToAdd = matchProductModel[1];
+        linkedProductModels.filter((linkedProduct) =>  linkedProduct.identifier !== productModelIdentifierToAdd);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:841]: linkedProductModels.filter((linkedProduct) =>  linkedProduct.identifier !== productModelIdentifierToAdd.identifier);

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js [hunk 9]
```diff
@@ -742,6 +809,47 @@ define([
...
+        [];
+
+      const linkedProducts = productAndProductModelIdentifiers.reduce((linkedProducts, identifier) => {
+        const matchProduct = identifier.match(/^product_(.*)$/);
+        if (!matchProduct) {
+          return linkedProducts;
+        }
+
+        const productIdentifierToAdd = matchProduct[1];
+        linkedProducts.filter((linkedProduct) =>  linkedProduct.identifier !== productIdentifierToAdd);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/associations.js:829]: linkedProducts.filter((linkedProduct) =>  linkedProduct.identifier !== productIdentifierToAdd.identifier);

**Pull request closed without merging**